ext/drmgr/drmgr.h

```diff
@@ -721,6 +721,72 @@ DR_EXPORT
 ptr_uint_t
 drmgr_reserve_note_range(size_t size);
 
+/***************************************************************************
+ * EMULATION
+ */
+
+/**
+ * TODO: Full doxygen comments will be done as part of upstream pull request,
```

Comment:
So your plan is to fill these in now, commit this, then add an iterator in a 2nd PR and change the sample to use it, and then change rep string expansion to use this in a 3rd PR?

Reply:
Yes, that is our preference and if that is acceptable to to you. The iterator change was something we had discussed and would ideally like to have done in this PR but other commitments mean it will have to be delayed by a week or so.

Reply:
#3112